core/services/run_executor.go

```diff
@@ -97,7 +98,7 @@ func (re *runExecutor) Execute(runID *models.ID) error {
 
 		validated = true
 
-		if err := re.store.ORM.SaveJobRun(&run); errors.Cause(err) == orm.ErrOptimisticUpdateConflict {
+		if err := re.store.ORM.SaveJobRun(&run); err == orm.ErrOptimisticUpdateConflict {
```

Comment:
Hmm I still think you want `errors.Cause` here?

Reply:
no SaveJobRun returns that error directly